spoken/forms.py

```diff
@@ -13,11 +13,34 @@ class KeywordSearchForm(forms.Form):
 
 class TutorialSearchForm(forms.Form):
     try:
+        all_tests = FossAvailableForTest.objects.filter(status=1)
+        test_list = []
+        for item in all_tests:
+            test_list.append(item.foss.foss)
+
         foss_list = TutorialResource.objects.filter(Q(status=1) | Q(status=2), language__name='English').values('tutorial_detail__foss__foss').annotate(
             Count('id')).order_by('tutorial_detail__foss__foss').values_list('tutorial_detail__foss__foss', 'id__count').distinct()
         choices = [('', '-- All Courses --'), ]
+
+        # segregate list for foss courses with and without tests
+        choices.append(['', ''])
+        choices.append(['', '-------------------------------------------------------------------'])
+        choices.append(['', '        FOSS Courses available for tests'])
+        choices.append(['', '-------------------------------------------------------------------'])
+        choices.append(['', ''])
         for foss_row in foss_list:
-            choices.append((str(foss_row[0]), str(foss_row[0]) + ' (' + str(foss_row[1]) + ')'))
+            if foss_row[0] in test_list:
+                choices.append((str(foss_row[0]), str(foss_row[0]) + ' (' + str(foss_row[1]) + ')'))
+
+        choices.append(['', ''])
+        choices.append(['', '-------------------------------------------------------------------'])
+        choices.append(['', '        FOSS Courses NOT available for tests'])
+        choices.append(['', '-------------------------------------------------------------------'])
+        choices.append(['', ''])
+        for foss_row in foss_list:
+            if foss_row[0] not in test_list:
+                choices.append((str(foss_row[0]), str(foss_row[0]) + ' (' + str(foss_row[1]) + ')'))
```

Comment:
We should be using `optgroup` to provide the categorization.

here is an example to do the django way: https://stackoverflow.com/a/17854288/782901